core/sail/lmdb/src/main/java/org/eclipse/rdf4j/sail/lmdb/TripleStore.java

```diff
@@ -779,13 +785,13 @@ void endTransaction(boolean commit) throws IOException {
 						E(mdb_txn_commit(writeTxn));
 					} finally {
 						recordCache = null;
-						readTxnRef.activate();
-						readTxnRef.lock().unlockWrite(stamp);
+						txnManager.activate();
+						txnManager.lock().unlockWrite(stamp);
```

Comment:
txnManager.activate() could potentially throw an exception so that the write lock is never unlocked.

Reply:
I'll wrap it in a try {...} finally {...} block.